File: my-workout-api/src/main/java/com/avernet/myworkoutapi/workout/WorkoutEntity.java
package com.avernet.myworkoutapi.workout;

import com.avernet.myworkoutapi.history.HistoryEntity;
import com.avernet.myworkoutapi.musclegroup.MuscleGroupEntity;
import com.avernet.myworkoutapi.user.UserEntity;
import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDate;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "workout")
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class WorkoutEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Long id;

    @Column()
    LocalDate date;

    @Column()
    Integer duration;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id")
    UserEntity user;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "muscle_group_id")
    MuscleGroupEntity muscleGroup;


    @OneToMany(mappedBy = "workout", fetch = FetchType.LAZY, cascade = CascadeType.ALL, orphanRemoval = true)
    List<HistoryEntity> histories = new ArrayList<>();
}

File: my-workout-api-nest/src/workout/workout.entity.ts
import { Column, Entity, JoinColumn, ManyToOne, OneToMany, PrimaryGeneratedColumn } from 'typeorm';
import { UserEntity } from '../user/user.entity';
import { MuscleGroupEntity } from '../muscle-group/muscle-group.entity';
import { HistoryEntity } from '../history/history.entity';

@Entity({ name: 'workout', schema: 'public' })
export class WorkoutEntity {
    @PrimaryGeneratedColumn('increment')
    id: number;

    @ManyToOne(() => UserEntity, user => user.workout, { onDelete: 'CASCADE' })
    @JoinColumn({ name: 'user_id' })
    user: UserEntity;

    @ManyToOne(() => MuscleGroupEntity, muscleGroup => muscleGroup.workout, { onDelete: 'CASCADE' })
    @JoinColumn({ name: 'muscle_group_id' })
    muscleGroup: MuscleGroupEntity;

    @Column({ type: 'timestamp' })
    date: Date;

    @Column()
    duration: number;

    @OneToMany(() => HistoryEntity, history => history.workout)
    history: HistoryEntity[];
}
